File: java/ecbenchmark/src/ecbenchmark/cec10/F02.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.cec10;

import ecbenchmark.function.Rastrigin;
import ecbenchmark.Function;
import ecbenchmark.operator.Shifted;

/**
 *
 * @author jcrada
 */
public class F02 extends CecFunction {

    private Function rastrigin;

    public F02(int dimensions) {
        super("F02", "Shifted Rastrigin's Function", dimensions,
                -5, 5, true, new CecRandom(2l), 0);
        double[] shift = CecMath.ShiftVector(dimensions, getMinimum(), getMaximum(), getRandomizer());
        rastrigin = new Shifted(shift, 0, new Rastrigin());
    }

    public double f(final double[] x) {
        return rastrigin.f(x);
    }
}

File: cpp/ecbenchmark/src/cec10/F02.cpp
#include "ecbenchmark/cec10/F02.h"
#include "ecbenchmark/cec10/CecMath.h"
#include "ecbenchmark/operator/Shifted.h"
#include "ecbenchmark/function/Rastrigin.h"

namespace ecb {
    namespace cec10 {

        F02::F02(int dimensions)
        : CecFunction("F02", "Shifted Rastrigin's Function", dimensions,
        -5, 5, true, 0, new CecRandom(2l)) {
            std::vector<scalar> shift = CecMath::ShiftVector(dimensions, minimumDomain(),
                    maximumDomain(), randomizer());
            rastrigin = new Shifted(shift, 0, new Rastrigin());
        }

        F02::~F02() {
            delete rastrigin;
        }

        scalar F02::f(const std::vector<scalar>& x) {
            return rastrigin->f(x);
        }
    }
}
